podcasts/SwiftUI/ScrollViewWithContentOffset.swift

```diff
@@ -0,0 +1,78 @@
+import SwiftUI
+
+/// `ScrollView` wrapper that allows views to be notified when the content offset changes by using
+/// `.onContentOffsetChange`
+///
+/// See the [Preview](x-source-tag://ScrollViewWithContentOffsetPreview) for example usage
+///
+struct ScrollViewWithContentOffset<Content: View>: View {
+    let content: () -> Content
+
+    typealias ContentOffsetHandler = (CGPoint) -> Void
+    @Namespace private var coordinateSpace
+    private let coordinator = Coordinator()
+
+    var body: some View {
+        ScrollView {
+            content().background(offsetObserverView)
+        }
+        .coordinateSpace(name: coordinateSpace)
+        .onPreferenceChange(ScrollOriginPreferenceKey.self) { origin in
+            coordinator.contentOffsetHandler?(origin)
+        }
+    }
+
+    /// Add a callback that is called when the content offset changes
+    func onContentOffsetChange(_ action: @escaping ContentOffsetHandler) -> some View {
```

Comment:
[nit] What do you think about grouping the `var` and declaring the `func` after them?